### DISCUSSION TITLE
The atomic operations in networkX?

### DISCUSSION THREAD

**Discussion by zhang-xh95** [networkx/networkx#6646] [Q&A] [answered]
Hi, I'm wondering if the operations that change graph, e.g., add_edge, is atomic?

**Comment by dschult** [comment]
Can you say what you mean by atomic?

**Comment by boothby** [comment, marked as answer]
I believe that this question is about thread safety, and the answer is no, we use structures such as python dicts which do not support atomic operations.

**Reply by zhang-xh95** [reply]
I got it. Thanks for your reply.
